src/class-wpml-vc-website-builder-translatable-nodes.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/**
+ * Class WPML_VC_Website_Builder_Translatable_Nodes
+ */
+class WPML_VC_Website_Builder_Translatable_Nodes implements IWPML_Page_Builders_Translatable_Nodes {
+
+	const SETTINGS_FIELD = 'settings';
+
+	/**
+	 * @var string
+	 */
+	private $settings_field;
+
+	/**
+	 * @var string
+	 */
+	private $type;
+
+	/** @var array */
+	private $nodes_to_translate;
+
+	public function __construct() {
+		$this->settings_field = self::SETTINGS_FIELD;
+		$this->type           = 'tag';
+	}
+
+	/**
+	 * @param string|int $node_id
+	 * @param array $settings
+	 *
+	 * @return WPML_PB_String[]
+	 */
+	public function get( $node_id, $settings ) {
+
+		if ( ! $this->nodes_to_translate ) {
+			$this->initialize_nodes_to_translate();
+		}
+
+		$strings = array();
+
+		foreach ( $this->nodes_to_translate as $node_type => $node_data ) {
+			if ( $this->conditions_ok( $node_data, $settings ) ) {
+				foreach ( $node_data['fields'] as $field ) {
+					$field_key = $field['field'];
+					if ( isset( $settings[ $field_key ] ) && trim( $settings[ $field_key ] ) ) {
```

Comment:
@adrianosferreira If all this code is in a common abstract base class then you can call `$this->get_setting( $settings, $field_key )` here. Then in the child class you add 
```
protected function get_setting( $setting, $feild_key ) {
   return $setting[ $field_key ];
}
```

And in BB

```
protected function get_setting( $setting, $feild_key ) {
   return $setting->$field_key;
}
```

Reply:
@brucepearson makes sense.

Reply:
@brucepearson please check the new commit here and also this new PR: https://github.com/OnTheGoSystems/wpml-page-builders/pull/3

I've done what you asked but didn't change Elementor and PB classes yet.